Sources/SwiftSyntax/Identifier.swift

```diff
@@ -0,0 +1,60 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// An abstraction for sanitized values on a token.
+public struct Identifier: Equatable, Hashable, Sendable {
+  /// The sanitized `text` of a token.
+  public var name: String {
+    String(syntaxText: raw.name)
+  }
+
+  @_spi(RawSyntax)
+  public let raw: RawIdentifier
+
+  public init?(_ token: TokenSyntax) {
+    guard case .identifier = token.tokenKind else {
+      return nil
+    }
+
+    self.raw = RawIdentifier(token.tokenView)
+  }
+}
+
+@_spi(RawSyntax)
+public struct RawIdentifier: Equatable, Hashable, Sendable {
+  public let name: SyntaxText
+
+  private let arena: RetainedSyntaxArena
+
+  @_spi(RawSyntax)
+  fileprivate init(_ raw: RawSyntaxTokenView) {
+    let backtick = SyntaxText(StaticString(stringLiteral: "`"))
```

Comment:
Is the `StaticString` needed here? Couldn’t this just be the following? I would expect the type checker to infer the string literal to be a static string.

```suggestion
    let backtick = SyntaxText("`")
```